static/app/views/starfish/components/detailPanel.tsx

```diff
@@ -20,8 +22,25 @@ type DetailState = {
   collapsed: boolean;
 };
 
+const SLIDEOUT_STORAGE_KEY = 'starfish-panel-slideout-direction';
+
+function getSlideOutDirection(): 'Right' | 'Bottom' {
+  const localValue = localStorage.getItem(SLIDEOUT_STORAGE_KEY) || 'Right';
+  if (localValue !== 'Bottom' && localValue !== 'Right') {
+    return 'Right';
+  }
+  return localValue;
+}
+
+function setSlideOutDirection(direction: 'Right' | 'Bottom') {
+  localStorage.setItem(SLIDEOUT_STORAGE_KEY, direction);
+}
+
```

Comment:
I think what you want here is `useLocalStorageState`:

```jsx
import {useLocalStorageState} from 'sentry/utils/useLocalStorageState';

...

const [slidePosition, setSlidePosition] = useLocalStorageState(SLIDEOUT_STORAGE_KEY, 'right');
```

then there should be no need for the `useEffect` down below.

Reply:
😱 Goddamn that's so much cleaner lol, I just grepped code for localStorage and copied 😛